### PR TITLE
[Serializer] feature: add the context key and add a link to the encoders in serializer to avoid duplication.

### LINKED ISSUE #9384: [Serializer] add a context key to return always as collection for XmlEncoder
https://github.com/symfony/symfony/pull/25369

### PR DESCRIPTION
**Pull request by Simperfit**
Fixes #9384 and #9383.

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

File: components/serializer.rst [hunk 1]
```diff
@@ -605,23 +605,7 @@ There are several types of normalizers available:
 Encoders
```
Comment (javiereguiluz) [inline components/serializer.rst]: Considering that we show all the information about `Normalizers` in this article ... I think the change should be the opposite: delete the `serializer/encoders.rst` article entirely and move its contents to this `Encoders` section.

Comment (Simperfit) [reply components/serializer.rst]: I will do

**Comment by Simperfit** [comment]
Status: Needs Review

**Comment by javiereguiluz** [comment]
Hamza, thanks for your work here. In addition to documenting missing things, you removed a small doc article, and we always love that! Cheers.

**Pull request merged**